UNSW-CEEM/tm_solarshift
# Fix stc bug - unfinished

A new layout (in trnsys) of solar thermal collector was created. The new layout is not working properly. For some reason, the aux heater is heating the whole time after it is turn on. The controller is not working properly. This seems to happen when it's called from tm_solarshift, not from Simulation Studio (trnsys GUI).
Actions to test (troubleshooting):
- copy the temperature control system from the RS layout (check the folder in data/trnsys_layout). Probably an old RS layout with MM would work better.
- check the models/trnsys.py file to check if there is not any variable that is being overwritten by the standard HW_tank. I checked this, but for some reason is not working.

During the updating of gas heaters for Melbourne I noticed an unexpected results for gas_instant. Its annual energy cost was way lower. It was due to a unit conversion in the tariff calculations. Originally the tariffs were calculated in AUD/MJ for gas_heaters. Now, it is in AUD/kWh for any heater. I had to update the unit handling of GasHeaterInstantaneous an update the timeseries.market.load_household_gas_rate() function. I found a bug and fix it (last commit).

I'll finish the stc bug soon.
